internal/imagedestination/impl/compat.go

```diff
@@ -99,3 +99,9 @@ func (c *Compat) PutSignatures(ctx context.Context, signatures [][]byte, instanc
 	}
 	return c.dest.PutSignaturesWithFormat(ctx, withFormat, instanceDigest)
 }
+
+// MatchesRequiredCompression validates if compression is required by the caller while selecting a blob, if it is required
+// then function performs a match against the compression requested by the caller and compression of exisiting blob.
+func (c *Compat) MatchesRequiredCompression(options private.TryReusingBlobOptions, info types.BlobInfo) bool {
+	return options.RequiredCompression == nil || (options.RequiredCompression.Name() == info.CompressionAlgorithm.Name())
```

Comment:
Note the part of https://github.com/containers/image/pull/2023#issuecomment-1624461020 about `info.CompressionAlgorithm` in general not being available…

I think that value should be in `TryReusingBlobOptions`, and then this can be just a
```go
func {CanReuse,OptionsAllowReusing}ExactBlob(options private.TryReusingBlobOptions) bool
```

Or, alternatively, `BlobMatchesRequiredCompression(options …, *compressiontypes.Algorithm) bool` (not requiring a `BlobInfo` would allow this to be used in the `CanSubstitute` case as well.